pipenv/utils/resolver.py

```diff
@@ -418,7 +418,7 @@ def get_deps_from_req(
                         err=True,
                     )
                 return constraints, locked_deps
-            constraints.add(req.constraint_line)
+            constraints[req.constraint_line] = None
             return constraints, locked_deps
         return constraints, locked_deps
```

Comment:
Shouldn't this return `constraints.keys(), locked_deps` or whatever calls `.keys()` on the constraints dictionary?

Reply:
It depends on how it used down the road I guess. If we are just iterating it, like a list then it is ok. Since, iterating is over keys only.

Reply:
@matteius You are right, it would be nicer if we convert this set to a list when we are returning it. This particular function is actually internal to this class, but the function get_metadata is public I will fix this ...

Reply:
`list[constraints]` is an list while `constraints.keys()` is an iterator.